entity-service/src/main/kotlin/com/egm/stellio/entity/web/EntityOperationHandler.kt

```diff
@@ -30,16 +25,18 @@ class EntityOperationHandler(
                 extractAndParseBatchOfEntities(it)
             }
             .map {
-                val existingEntities = validationUtils.getExistingEntities(it)
-                val newEntities = validationUtils.getNewEntities(it)
-                val validEntities = validationUtils.getValidEntities(newEntities)
-                Triple(existingEntities, newEntities, validEntities)
-            }
-            .map {
-                    entityService.processBatchOfEntities(it.first, it.second, it.third)
+                val (existingEntities, newEntities) = entityOperationService.splitEntitiesByExistence(it)
+                val batchOperationResult = entityOperationService.create(newEntities)
+
+                batchOperationResult.errors.addAll(
+                    existingEntities.map {
+                        BatchEntityError(it.id, arrayListOf("Entity already exists"))
+                    })
+
+                batchOperationResult
             }
             .flatMap {
                 ok().body(BodyInserters.fromValue(it))

```

Comment:
@bobeal should I implement the new response codes in this PR? there are already lots of changes

Reply:
Let's stick to version 1.2.1 of the spec for now. It will avoid mixing versions and we'll probably make a 1.3.1 session later.